packages/frontend/src/system-integration/notifications.ts

```diff
@@ -81,11 +115,50 @@ async function showNotification(
     try {
       const notificationInfo =
         await BackendRemote.rpc.getMessageNotificationInfo(accountId, messageId)
-      const { chatName, summaryPrefix, summaryText } = notificationInfo
+      let summaryPrefix = notificationInfo.summaryPrefix
+      const summaryText = eventText ?? notificationInfo.summaryText
+      const chatName = notificationInfo.chatName
+      let icon = getNotificationIcon(notificationInfo)
+      if (isWebxdcInfo && chatId === -1) {
```

Comment:
And `chatId` doesn't seem to get overridden even if it's `-1`. Is it fine to pass `chatId === -1` to `runtime.showNotification()`? Doesn't seem so at a glance.

I agree, it's probably better just to add `chatId` to the event.

Reply:
> I agree, it's probably better just to add chatId to the event.

Actually, it's present on `notificationInfo`, can be obtained from there.

Reply:
Oh I see. This condition is never even true, because `chatId` is never actually -1 here, because it's set inside of `flushNotifications`, in that new `for await` loop.

Reply:
> This condition is never even true, because chatId is never actually -1 here, 

right - the chatId condition is wrong and not needed here

> So yeah, I would remove that loop, and just get chatId from notificationInfo unconditionally

chatId is needed in flushNotifications to find mutedChats which should not show notifications

> showNotification doesn't need the chatId parameter even

not sure which "showNotification" you mean. frontend showNotification passes the chatId to runtime.showNotification which passes it to ipcMain.handle('notifications.show', showNotification) which needs the chatId for the callbacks

Reply:
> chatId is needed in flushNotifications to find mutedChats which should not show notifications

Whoops, forgot about that.

> not sure which "showNotification" you mean.

I mean the bare function defined in this file.

> frontend showNotification passes the chatId to runtime.showNotification which passes it to ipcMain.handle('notifications.show', showNotification) which needs the chatId for the callbacks

Yes, but inside of `showNotification` `chatId` is available on `notificationInfo`.